snap/info.go

```diff
@@ -1010,14 +1010,11 @@ func (app *AppInfo) ServiceFile() string {
 	return filepath.Join(dirs.SnapServicesDir, app.ServiceName())
 }
 
-// Env returns the app specific environment overrides
-func (app *AppInfo) Env() []string {
-	appEnv := app.Snap.Environment.Copy()
-	for _, k := range app.Environment.Keys() {
-		appEnv.Set(k, app.Environment.Get(k))
-	}
-
-	return envFromMap(appEnv)
+// EnvironmentOverrides returns the app-specific environment overrides.
+func (app *AppInfo) EnvironmentOverrides() *strutil.EnvironmentDelta {
+	delta := &strutil.EnvironmentDelta{OrderedMap: *app.Snap.Environment.Copy()}
```

Comment:
why the Copy, I would hope EnvironmentDelta to be defined as immutable, either way, it probably needs a New* and not direct construction as here if we are worried

Reply:
Effectively because Merge. We can make Merge a function that takes a number of deltas and do away with Copy.

Reply:
per discussion this actually needs to return []osutil.ExpandableEnv

Reply:
Done